### PR TITLE
Core/Packets: convert CMSG_ENUM_CHARACTERS and SMSG_ENUM_CHARACTERS_R…

### PR DESCRIPTION
**Pull request by Ghaster**
…ESULT to packet class

Defined all fields in SMSG_ENUM_CHARACTERS_RESULT. Tested ingame with multiple characters, moving position of characters and logging in to world.

### CODE REVIEW

File: src/server/game/Server/Packets/CharacterPackets.h [hunk 1]
```diff
@@ -53,23 +53,23 @@ namespace WorldPackets
...
+                int32 ZoneID          = 0;
+                int32 MapID           = 0;
                 TaggedPosition<Position::XYZ> PreloadPos;
                 ObjectGuid GuildGUID;
-                uint32 Flags             = 0; ///< Character flag @see enum CharacterFlags
-                uint32 Flags2            = 0; ///< Character customization flags @see enum CharacterCustomizeFlags
-                bool FirstLogin      = false;
+                uint32 Flags          = 0; ///< Character flag @see enum CharacterFlags
+                uint32 Flags2         = 0; ///< Character customization flags @see enum CharacterCustomizeFlags
+                bool FirstLogin       = false;
```
Comment (Ovahlord) [inline src/server/game/Server/Packets/CharacterPackets.h]: actually you can remove these spaces entirely to follow the standards of the other packet class member files

Comment (Warpten) [reply src/server/game/Server/Packets/CharacterPackets.h]: Order members by size, larger first (u64 > u32 > u16 > u8 > string)

Comment (Ghaster) [reply src/server/game/Server/Packets/CharacterPackets.h]: - Not entirely sure what you mean by this. Is it related to ZoneID and MapID being 32 bits in header, but stored as 16 bits in the DB? 
- I agree, however there is no alternative with the readability of TaggedPosition. (The only one would be G3D::Vector3, but TaggedPosition was implemented with the goal of removing G3D from packet headers.)
- I'll let you wrestle that out with Elon. Winner gets to decide whether there should be empty lines or not.

Comment (Ovahlord) [reply src/server/game/Server/Packets/CharacterPackets.h]: He means you should order the members by type size for optimal performance. (bool, uint8, uint16, uint32 etc)
About the spacings that is for you to decide. warpten and me have some different tastes when it comes to codestyle so just pick your favourite here.

Comment (Ghaster) [reply src/server/game/Server/Packets/CharacterPackets.h]: Alright should be sorted, not sure if `PreloadPos` and `VisualItems` are located correctly.

**Comment by Warpten** [comment]
Good size-wise. `PreloadPos` is really `float[4]`, so right there is OK. `VisualItems` is effectively `uint32[3]`. It doesn't matter where you put it, so long as it's somewhere that doesn't add padding (in the middle of four `uint8`s, for example).

As far as `TaggedPosition` goes, my point is that you can just use `Position`, aka drop the tag, because you're not using it. This makes zero changes to the structure size but #include-what-you-use.

I realize most of these remarks are nitpick, though, so /shrug

**Comment by Ghaster** [comment]
Nitpicks are most welcome. I have plans of converting more packets into this system. Having some "guidelines" for how they should look makes it bit easier, instead of declaring members at random in the header.

**Comment by Ovahlord** [comment]
any final words before I push the shiny button?

**Comment by Ghaster** [comment]
go right ahead

**Pull request merged**